R/tax_unique.R

```diff
@@ -135,105 +132,55 @@ tax_unique <- function(occdf = NULL, binomial = NULL, species = NULL,
   if (!is.null(binomial) && !(binomial %in% colnames(occdf))) {
     stop("`occdf` does not contain column name provided to `binomial`")
   }
-  if (!is.null(binomial)) {
-    colnames(occdf)[colnames(occdf) == binomial] <-
-      "binomial"
-  }
 
   if (!is.null(species) && !(species %in% colnames(occdf))) {
     stop("`occdf` does not contain column name provided to `species`")
   }
-  if (!is.null(species)) {
-    colnames(occdf)[colnames(occdf) == species] <-
-      "species"
-  }
 
   if (!is.null(genus) && !(genus %in% colnames(occdf))) {
     stop("`occdf` does not contain column name provided to `genus`")
   }
-  if (!is.null(genus)) {
-    colnames(occdf)[colnames(occdf) == genus] <-
-      "genus"
-  }
 
-  if (is.null(family)) {
-    stop("`family` argument is missing. Provide column name of family names.")
-  }
-  if (!(family %in% colnames(occdf))) {
-    stop("`occdf` does not contain column name provided to `family`")
+  higher_args <- list(...)
+  higher_names <- names(higher_args)
+  higher_cols <- unname(unlist(higher_args))
+  if (length(higher_args) == 0) {
+    stop("At least one higher taxonomic level must be supplied (e.g. family)")
```

Comment:
```suggestion
    stop("At least one higher taxonomic level must be supplied (e.g. `family`)")
```
Maybe put this in backticks similar to the other error messages to help indicate an additional argument is needed?